plugins/event_source/eventstream.py

```diff
@@ -0,0 +1,293 @@
+            aiohttp.ClientResponseError: If the server responds with an HTTP status code that indicates an error.
+            aiohttp.ClientConnectionError: If there is a problem with the underlying TCP connection.
+            aiohttp.ClientTimeoutError: If the connection times out.
+        """
+        event_type_filter = "" if not self.include_event_types else f"&eventType={','.join(self.include_event_types)}"
+        offset_filter = f"&offset={self.offset}"
+
+        kwargs = {
+            "url": f"{self.data_feed}{offset_filter}{event_type_filter}",
+            "headers": {
+                "Authorization": f"Token {self.token}",
+            },
+            "raise_for_status": True,
+            "timeout": aiohttp.ClientTimeout(total=float(self.refresh_interval)),
+        }
+
+        session = aiohttp.ClientSession()
+        self.spigot: aiohttp.ClientResponse = await session.get(**kwargs)
+        logger.info("Successfully opened stream %s:%s", self.stream_name, self.partition)
+        return self.spigot
+
+    async def stream_events(self, exclude_event_types: List[str]) -> AsyncGenerator[Dict[str, Any], None]:
+        """
+        Asynchronously generate events from the CrowdStrike Falcon Event Stream API.
+
+        This method opens a stream to the Falcon API, iterates over the lines in the stream, and decodes and yields each event.
+        It automatically refreshes the client token and reopens the stream if the token has expired.
+
+        Args:
+            exclude_event_types (List[str]): A list of event types to be excluded from the stream.
+
+        Yields:
+            Dict[str, Any]: A dictionary containing the event data from the Falcon API and a count of event types seen so far.
+
+        Raises:
+            aiohttp.ClientResponseError: If the server responds with an HTTP status code that indicates an error.
+            ValueError: If client authentication fails during the token refresh process.
+        """
+        # Open the stream
+        await self.open_stream()
+        # Asynchronously iterate over the lines in the stream
+        async for line in self.spigot.content:
+            # Decode the line from bytes to a string and remove trailing whitespace
+            line = line.decode().rstrip()
+            if line:
+                # Load the event as a JSON object
+                json_event = json.loads(line)
+                event_type = json_event["metadata"]["eventType"]
+                self.offset = json_event["metadata"]["offset"]
+                # If the event is valid, yield it
+                if self.is_valid_event(event_type, exclude_event_types):
+                    # If INFO or DEBUG, update the event_type_count dict and yield it
+                    if logger.level <= logging.INFO:
+                        self.event_type_count[event_type] = self.event_type_count.get(event_type, 0) + 1
+                        yield dict(falcon=json_event, eventTypeCount=self.event_type_count)
```

Comment:
@carlosmmatos remove due to issue with multiple streams and not really being an ask right now